packages/helpwave_service/lib/src/user_model.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:flutter/material.dart';
+import 'package:helpwave_service/src/user_preference.dart';
+
+/// Model for saving user
+///
+/// Notifies about changes of the user on registration or update
+class UserModel extends ChangeNotifier {
+  bool _isInitialized = false;
+  String _user = "";
```

Comment:
The User Service should be written for every app or we can think about generalizing this and save arbitrary object that are serializable.
E.g. make this class generic expecting the generic to implement fromJson and toJson and then use a similar logic as here (also pass a key for the preferences)

Reply:
> Change this to a User object and only convert to json in the set and get preferences methods

Do you men implementing the user class in the service package?

Reply:
No, I meant implementing a specific service in each app as the information required for a user might differ and if we were to generalize a user class this would need a more indepth analysis and should propbably also be done in a seperate issue